packages/metadata-common/src/MetadataEditor.tsx

```diff
@@ -100,6 +101,22 @@ export class MetadataEditor extends ReactWidget {
         if (!this.name) {
           this.title.label = `New ${this.schemaDisplayName}`;
         }
+        // Find categories of all schema properties
+        this.categories = { _noCategory: [] };
+        for (const schemaProperty in this.schema) {
+          const cat =
+            this.schema[schemaProperty].uihints &&
+            this.schema[schemaProperty].uihints.category;
+          if (!cat) {
+            this.categories['_noCategory'].push(schemaProperty);
+          } else if (
+            Object.prototype.hasOwnProperty.call(this.categories, cat)
```

Comment:
is there a reason we're using `Object.prototype` here and not just `if (this.categories[cat])` to check existence in the dict?